### ISSUE TITLE
pr2_head_action is failing to build on the farm (hydro)

### ISSUE THREAD

**Issue by tfoote** [PR2/pr2_controllers#370]
I expect it's an upstream problem but I'm not sure where so filing here for now. 

This is the log: http://jenkins.ros.org/view/HbinP32/job/ros-hydro-pr2-head-action_binarydeb_precise_i386/32/console

```
[  0%] Built target actionlib_generate_messages_lisp
make -f CMakeFiles/actionlib_generate_messages_py.dir/build.make CMakeFiles/actionlib_generate_messages_py.dir/depend
make[4]: Entering directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
cd /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006 /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006 /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu/CMakeFiles/actionlib_generate_messages_py.dir/DependInfo.cmake --color=
Scanning dependencies of target actionlib_generate_messages_py
make[4]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
make -f CMakeFiles/actionlib_generate_messages_py.dir/build.make CMakeFiles/actionlib_generate_messages_py.dir/build
make[4]: Entering directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
make[4]: Nothing to be done for `CMakeFiles/actionlib_generate_messages_py.dir/build'.
make[4]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
/usr/bin/cmake -E cmake_progress_report /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu/CMakeFiles 
[  0%] Built target actionlib_generate_messages_py
make -f CMakeFiles/pr2_head_action.dir/build.make CMakeFiles/pr2_head_action.dir/depend
make[4]: Entering directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
cd /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006 /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006 /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu/CMakeFiles/pr2_head_action.dir/DependInfo.cmake --color=
Scanning dependencies of target pr2_head_action
make[4]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
make -f CMakeFiles/pr2_head_action.dir/build.make CMakeFiles/pr2_head_action.dir/build
make[4]: Entering directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
/usr/bin/cmake -E cmake_progress_report /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu/CMakeFiles 4
[100%] Building CXX object CMakeFiles/pr2_head_action.dir/src/pr2_point_frame.cpp.o
/usr/lib/ccache/c++   -DROS_PACKAGE_NAME=\"pr2_head_action\" -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security  -I/usr/include/eigen3 -I/opt/ros/hydro/include    -o CMakeFiles/pr2_head_action.dir/src/pr2_point_frame.cpp.o -c /tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/src/pr2_point_frame.cpp
Linking CXX executable devel/lib/pr2_head_action/pr2_head_action
/usr/bin/cmake -E cmake_link_script CMakeFiles/pr2_head_action.dir/link.txt --verbose=1
/usr/lib/ccache/c++   -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security     -L CMakeFiles/pr2_head_action.dir/src/pr2_point_frame.cpp.o  -o devel/lib/pr2_head_action/pr2_head_action -rdynamic -lboost_thread-mt -lpthread /opt/ros/hydro/lib/liborocos-kdl.so.1.2.0 /opt/ros/hydro/lib/libcpp_common.so /opt/ros/hydro/lib/libroscpp_serialization.so /opt/ros/hydro/lib/librostime.so -lboost_date_time-mt -lboost_system-mt -lboost_thread-mt -lpthread /opt/ros/hydro/lib/libroscpp.so -lboost_signals-mt -lboost_filesystem-mt /opt/ros/hydro/lib/librosconsole.so -lboost_regex-mt -llog4cxx /opt/ros/hydro/lib/libxmlrpcpp.so /opt/ros/hydro/lib/libmessage_filters.so /opt/ros/hydro/lib/libtf.so /opt/ros/hydro/lib/libtf2_ros.so /opt/ros/hydro/lib/libactionlib.so /opt/ros/hydro/lib/libtf2.so /opt/ros/hydro/lib/libconsole_bridge.so /opt/ros/hydro/lib/libtf_conversions.so /opt/ros/hydro/lib/libkdl_conversions.so /opt/ros/hydro/lib/libkdl_parser.so -ltinyxml /opt/ros/hydro/lib/liburdf.so /opt/ros/hydro/lib/liburdfdom_sensor.so /opt/ros/hydro/lib/liburdfdom_model_state.so /opt/ros/hydro/lib/liburdfdom_model.so /opt/ros/hydro/lib/liburdfdom_world.so /opt/ros/hydro/lib/librosconsole_bridge.so -lpthread /opt/ros/hydro/lib/libcpp_common.so /opt/ros/hydro/lib/libroscpp_serialization.so /opt/ros/hydro/lib/librostime.so -lboost_date_time-mt -lboost_system-mt -lpthread /opt/ros/hydro/lib/libroscpp.so -lboost_signals-mt -lboost_filesystem-mt /opt/ros/hydro/lib/librosconsole.so -lboost_regex-mt -llog4cxx /opt/ros/hydro/lib/libxmlrpcpp.so /opt/ros/hydro/lib/libmessage_filters.so /opt/ros/hydro/lib/libtf.so /opt/ros/hydro/lib/libtf2_ros.so /opt/ros/hydro/lib/libactionlib.so /opt/ros/hydro/lib/libtf2.so /opt/ros/hydro/lib/libconsole_bridge.so /opt/ros/hydro/lib/libtf_conversions.so /opt/ros/hydro/lib/libkdl_conversions.so /opt/ros/hydro/lib/libkdl_parser.so -ltinyxml /opt/ros/hydro/lib/liburdf.so /opt/ros/hydro/lib/liburdfdom_sensor.so /opt/ros/hydro/lib/liburdfdom_model_state.so /opt/ros/hydro/lib/liburdfdom_model.so /opt/ros/hydro/lib/liburdfdom_world.so /opt/ros/hydro/lib/librosconsole_bridge.so -Wl,-rpath,/opt/ros/hydro/lib: 
/usr/lib/gcc/i686-linux-gnu/4.6/../../../i386-linux-gnu/crt1.o: In function `_start':
(.text+0x18): undefined reference to `main'
collect2: ld returned 1 exit status
make[4]: *** [devel/lib/pr2_head_action/pr2_head_action] Error 1
make[4]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
make[3]: *** [CMakeFiles/pr2_head_action.dir/all] Error 2
make[3]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006/obj-i686-linux-gnu'
dh_auto_build: make -j1 returned exit code 2
make[1]: *** [override_dh_auto_build] Error 2
make[1]: Leaving directory `/tmp/buildd/ros-hydro-pr2-head-action-1.10.6-0precise-20131206-0006'
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
E: Failed autobuilding of package
```

It was triggered as follows for the failing build: http://jenkins.ros.org/view/HbinP32/job/ros-hydro-pr2-head-action_binarydeb_precise_i386/32/

```
Started by upstream project ros-hydro-orocos-kdl_binarydeb_precise_i386 build number 16
originally caused by:

Started by upstream project ros-hydro-orocos-kdl_sourcedeb build number 4
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-kdl-parser_binarydeb_precise_i386 build number 88
originally caused by:

Started by upstream project ros-hydro-orocos-kdl_binarydeb_precise_i386 build number 16
originally caused by:
Started by upstream project ros-hydro-orocos-kdl_sourcedeb build number 4
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-tf-conversions_binarydeb_precise_i386 build number 33
originally caused by:

Started by upstream project ros-hydro-kdl-conversions_binarydeb_precise_i386 build number 58
originally caused by:
Started by upstream project ros-hydro-orocos-kdl_binarydeb_precise_i386 build number 16
originally caused by:
Started by upstream project ros-hydro-orocos-kdl_sourcedeb build number 4
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-kdl-parser_binarydeb_precise_i386 build number 89
originally caused by:

Started by upstream project ros-hydro-kdl-parser_sourcedeb build number 19
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-cmake-modules_binarydeb_precise_i386 build number 8
originally caused by:
Started by upstream project ros-hydro-cmake-modules_sourcedeb build number 2
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-urdf_binarydeb_precise_i386 build number 45
originally caused by:
Started by upstream project ros-hydro-urdf_sourcedeb build number 19
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-cmake-modules_binarydeb_precise_i386 build number 8
Started by upstream project ros-hydro-urdf-parser-plugin_binarydeb_precise_i386 build number 24
originally caused by:
Started by upstream project ros-hydro-urdf-parser-plugin_sourcedeb build number 9
originally caused by:
Started by user Internal
```

**Comment by tfoote** [comment]
This is a major regression, breaking 40+ packages. 
This package hasn't changed in months. 

New releases before this build appear to include: 
kdl_parser, orocos_kdl, cmake_modules, urdf, urdf_parser 

@isucan @smits @wjwwood might any of your changes have caused this? 

The file it is compiling definitely has a main: https://github.com/PR2/pr2_controllers/blob/hydro-devel/pr2_head_action/src/pr2_point_frame.cpp#L531

**Comment by wjwwood** [comment]
The new `cmake_modules` package just added a new `FindTBB.cmake` module and some documentation changes. Neither this package nor any of its dependencies use `TBB` so I would be surprised if the release of `cmake_modules` had anything to do with it.

**Comment by tfoote** [comment]
robot_state_publisher failed at the same time: http://jenkins.ros.org/view/HbinP32/job/ros-hydro-robot-state-publisher_binarydeb_precise_i386/41/ 

```
Started by upstream project ros-hydro-kdl-parser_binarydeb_precise_i386 build number 88
originally caused by:

Started by upstream project ros-hydro-orocos-kdl_binarydeb_precise_i386 build number 16
originally caused by:
Started by upstream project ros-hydro-orocos-kdl_sourcedeb build number 4
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-tf-conversions_binarydeb_precise_i386 build number 33
originally caused by:

Started by upstream project ros-hydro-kdl-conversions_binarydeb_precise_i386 build number 58
originally caused by:
Started by upstream project ros-hydro-orocos-kdl_binarydeb_precise_i386 build number 16
originally caused by:
Started by upstream project ros-hydro-orocos-kdl_sourcedeb build number 4
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-kdl-parser_binarydeb_precise_i386 build number 89
originally caused by:

Started by upstream project ros-hydro-kdl-parser_sourcedeb build number 19
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-cmake-modules_binarydeb_precise_i386 build number 8
originally caused by:
Started by upstream project ros-hydro-cmake-modules_sourcedeb build number 2
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-urdf_binarydeb_precise_i386 build number 45
originally caused by:
Started by upstream project ros-hydro-urdf_sourcedeb build number 19
originally caused by:
Started by user Internal
Started by upstream project ros-hydro-cmake-modules_binarydeb_precise_i386 build number 8
Started by upstream project ros-hydro-urdf-parser-plugin_binarydeb_precise_i386 build number 24
originally caused by:
Started by upstream project ros-hydro-urdf-parser-plugin_sourcedeb build number 9
originally caused by:
Started by user Internal
```

Slightly different error: http://jenkins.ros.org/view/HbinP32/job/ros-hydro-robot-state-publisher_binarydeb_precise_i386/41/console

```
[ 50%] Built target robot_state_publisher_solver
make -f CMakeFiles/robot_state_publisher.dir/build.make CMakeFiles/robot_state_publisher.dir/depend
make[4]: Entering directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu'
cd /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005 /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005 /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu/CMakeFiles/robot_state_publisher.dir/DependInfo.cmake --color=
Scanning dependencies of target robot_state_publisher
make[4]: Leaving directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu'
make -f CMakeFiles/robot_state_publisher.dir/build.make CMakeFiles/robot_state_publisher.dir/build
make[4]: Entering directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu'
/usr/bin/cmake -E cmake_progress_report /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu/CMakeFiles 4
[ 75%] Building CXX object CMakeFiles/robot_state_publisher.dir/src/joint_state_listener.cpp.o
/usr/lib/ccache/c++   -DROS_PACKAGE_NAME=\"robot_state_publisher\" -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security  -isystem /usr/include/eigen3 -I/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/include -I/opt/ros/hydro/include    -o CMakeFiles/robot_state_publisher.dir/src/joint_state_listener.cpp.o -c /tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/src/joint_state_listener.cpp
Linking CXX executable devel/lib/robot_state_publisher/robot_state_publisher
/usr/bin/cmake -E cmake_link_script CMakeFiles/robot_state_publisher.dir/link.txt --verbose=1
/usr/lib/ccache/c++   -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security     CMakeFiles/robot_state_publisher.dir/src/joint_state_listener.cpp.o  -o devel/lib/robot_state_publisher/robot_state_publisher -rdynamic devel/lib/librobot_state_publisher_solver.so -llog4cxx /opt/ros/hydro/lib/libroscpp.so -lpthread -lboost_signals-mt -lboost_filesystem-mt -lboost_system-mt /opt/ros/hydro/lib/libcpp_common.so /opt/ros/hydro/lib/libroscpp_serialization.so /opt/ros/hydro/lib/librostime.so -lboost_date_time-mt -lboost_thread-mt /opt/ros/hydro/lib/librosconsole.so -lboost_regex-mt -llog4cxx /opt/ros/hydro/lib/libxmlrpcpp.so /opt/ros/hydro/lib/libtf.so /opt/ros/hydro/lib/libmessage_filters.so /opt/ros/hydro/lib/libtf2_ros.so /opt/ros/hydro/lib/libactionlib.so /opt/ros/hydro/lib/libtf2.so /opt/ros/hydro/lib/libconsole_bridge.so /opt/ros/hydro/lib/libtf_conversions.so /opt/ros/hydro/lib/libkdl_conversions.so /opt/ros/hydro/lib/libkdl_parser.so -ltinyxml /opt/ros/hydro/lib/liburdf.so /opt/ros/hydro/lib/liburdfdom_sensor.so /opt/ros/hydro/lib/liburdfdom_model_state.so /opt/ros/hydro/lib/liburdfdom_model.so /opt/ros/hydro/lib/liburdfdom_world.so /opt/ros/hydro/lib/librosconsole_bridge.so -Wl,-rpath,/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu/devel/lib:/opt/ros/hydro/lib: 
/usr/bin/ld: CMakeFiles/robot_state_publisher.dir/src/joint_state_listener.cpp.o: undefined reference to symbol 'KDL::Tree::Tree(std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: note: 'KDL::Tree::Tree(std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)' is defined in DSO /opt/ros/hydro/lib/liborocos-kdl.so.1.2 so try adding it to the linker command line
/opt/ros/hydro/lib/liborocos-kdl.so.1.2: could not read symbols: Invalid operation
collect2: ld returned 1 exit status
make[4]: *** [devel/lib/robot_state_publisher/robot_state_publisher] Error 1
make[4]: Leaving directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu'
make[3]: *** [CMakeFiles/robot_state_publisher.dir/all] Error 2
make[3]: Leaving directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005/obj-i686-linux-gnu'
dh_auto_build: make -j1 returned exit code 2
make[1]: *** [override_dh_auto_build] Error 2
make[1]: Leaving directory `/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131206-0005'
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

**Comment by ahendrix** [comment]
It looks like orocos_kdl finally fixed how they're exporting libraries.

**Comment by wjwwood** [comment]
I ran bloom for you @trainman419, I also added you to the PR2 team on ros-gbp for future releases.

**Comment by ahendrix** [comment]
Confirmed fixed; builds of 1.10.7 are succeeding now.
